tokenizers/src/pre_tokenizers/metaspace.rs

```diff
@@ -44,6 +95,20 @@ impl Metaspace {
             replacement,
             str_rep: replacement.to_string(),
             add_prefix_space,
+            prepend_scheme: PrependScheme::Always, // always prepend for legacy purpose
+        }
+    }
+
+    pub fn new_with_prepend_scheme(
+        replacement: char,
+        add_prefix_space: bool,
+        prepend_scheme: PrependScheme,
+    ) -> Self {
+        Self {
+            replacement,
+            str_rep: replacement.to_string(),
+            add_prefix_space,
+            prepend_scheme,
```

Comment:
we could do `perpend_scheme.into()` and allow passing `<Into PrependScheme>` as type

Reply:
I don't think we should. Real type is better.